game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/retreat/Retreater.java

```diff
@@ -0,0 +1,32 @@
+package games.strategy.triplea.delegate.battle.steps.retreat;
+
+import games.strategy.engine.data.Change;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.delegate.battle.MustFightBattle;
+import java.util.Collection;
+import java.util.Map;
+
+interface Retreater {
+
+  enum RetreatLocation {
+    SAME_TERRITORY,
+    OTHER_TERRITORY;
+  }
+
+  Collection<Unit> getRetreatUnits();
+
+  Collection<Territory> getPossibleRetreatSites(Collection<Unit> retreatUnits);
+
+  String getQueryText();
+
+  MustFightBattle.RetreatType getRetreatType();
+
+  Map<RetreatLocation, Collection<Unit>> splitRetreatUnits(Collection<Unit> retreatUnits);
+
+  Change extraRetreatChange(Territory retreatTo, Collection<Unit> retreatUnits);
```

Comment:
I wonder if we are mixing responsibilities with this "extraRetreatChange" API. The retreat code probably should not know much about creating a change object. 

Suggestion: have the caller check the battle type, if amphib then the change is 'empty' or do not even add it to the change stack, otherwise inline the implementation code into the caller.